src/Brick/ReflexMain.hs

```diff
@@ -157,37 +159,51 @@ brickWrapper shouldHaltE widgetDyn cursorDyn attrDyn = do
             $ ((io >>= resultH) `finally` restartH ())
         return resultEvent
 
+  rsRef <- liftIO $ newIORef initialRS
+
+  let startOrSuspend :: These () () -> Maybe (Vty, IO ()) -> R.PushM t (Maybe (Vty, IO ()))
+      startOrSuspend = \case
+        This{} -> \_ -> do
+          widgetStack  <- R.sample $ R.current widgetDyn
+          chooseCursor <- R.sample $ R.current cursorDyn
+          attrs        <- R.sample $ R.current attrDyn
+          vty <- liftIO $ do
+            x <- mkVty defaultConfig
+            return x
+          liftIO $ do
+            renderState <- readIORef rsRef
+            (renderState', _) <-
+              render vty widgetStack chooseCursor attrs renderState
```

Comment:
is this code duplication necessary, including the `sample`s above? Would it not be possible to just change the definition of `refreshE` below so that it fires also on `initStateDyn` updates (i.e. remove `mState <- R.sample $ R.current initStateDyn` and use `mState <- initStateDyn` like for the other dynamics)? I might be missing something about ordering guarantees here and have not tested this, but in my brain this works :p